code/game/machinery/porta_turret/portable_turret.dm

```diff
@@ -1052,28 +1054,25 @@
 	else
 		cooldown = world.time + cooldown_duration
 		warned = FALSE
-		INVOKE_ASYNC(src, /obj/machinery/manned_turret/.proc/volley)
+		volley(user)
 
-/obj/machinery/manned_turret/proc/volley()
+/obj/machinery/manned_turret/proc/volley(mob/user)
 	target_turf = get_turf(target)
 	for(var/i in 1 to number_of_shots)
-		addtimer(CALLBACK(src, /obj/machinery/manned_turret/.proc/fire_helper), i*rate_of_fire)
+		addtimer(CALLBACK(src, /obj/machinery/manned_turret/.proc/fire_helper, user), i*rate_of_fire)
 
 
-/obj/machinery/manned_turret/proc/fire_helper()
+/obj/machinery/manned_turret/proc/fire_helper(mob/user)
 	if(!src)
 		return
 	var/turf/targets_from = get_turf(src)
 	if(QDELETED(target))
 		target = target_turf
 	var/obj/item/projectile/P = new projectile_type(targets_from)
-	P.current = targets_from
-	P.starting = targets_from
-	P.firer = src
+	P.firer = user
 	P.original = target
 	playsound(src, 'sound/weapons/Gunshot_smg.ogg', 75, 1)
-	P.yo = target.y - targets_from.y + rand(-1,1)
-	P.xo = target.x - targets_from.x + rand(-1,1)
+	P.preparePixelProjectile(target, target_turf, user, mouseparams, rand(-10, 10))

```

Comment:
Reason this is DNM is I don't have time to test this projectile code right now so don't actually merge this fix until I can do that or someone else can do that, 90% sure this will work but I don't want to break this further.